### DISCUSSION TITLE
Feedbacks on the folder/data structure

### DISCUSSION THREAD

**Discussion by bsantosh909** [Know-Nepal/borderlines#2] [Ideas]
Hi Guys 👋 

We wanted to discuss with everyone about the folder and data structure that should be used for this repository.
Any feedbacks are welcome.

## Data to be included in the repository
- Provinces
- Districts
- Cities

Our plan is to initially include only the static data (data that doesn't change over the period of time) in the repository. This is because having dynamic data requires lots of maintenance work.

Regarding the file format we have a poll open at https://github.com/Know-Nepal/borderlines/discussions/1. Make sure to share your thoughts on that as well.

## Data format
We are thinking that for all the entities mentioned above, we can have a single file with data in it. (_Cities file might be a bit bulky though_).

Also for every top level entity we will add a unique ID. For e.g Bagmati province will have ID `id: province-3`.
And in district object we will have something like `provinceId: province-3` so that we can merge the two entities if necessary.

Basically like SQL's primary key, foreign key and join query.

Do let us know if you think this is correct or we should just include the name and few other details in the object itself.

**Comment by thenishantsapkota** [comment]
I like the idea you're proposing!
But the city file will get bulky real soon.
Let's seperate cities for each state inside of a `Cities` folder.
It'll be easier to manage?
Let me know your thoughts on this.

**Reply by twlite** [reply]
Yeah, something like that would be better

```
~/province
  /district
    /city
      /...
```

**Reply by ofcoursenp** [reply]
Yeah

**Reply by bsantosh909** [reply]
@archaeopteryx1 if we go with this format, how are we gonna handle it once we build the API?

**Reply by twlite** [reply]
> @archaeopteryx1 if we go with this format, how are we gonna handle it once we build the API?

That type of approach feels the best for storing the data. We could place a file named `main.json` in each dir which would define paths to other data inside that dir (something like github api, where it returns a url to pr, issues etc instead of returning whole data at root)

so the design would be something like this:
`GET /` <--- will list out all provinces (partial)
`GET /:province` <--- will return full province related data and partial district information with reference to its cities
`GET /:province/:city` <--- will return city info and ref. to other child info and so on

However, we can pick another design plan if this is too complicated to implement. Perhaps a graphql api would be better at this job.

**Reply by bsantosh909** [reply]
Sounds interesting.
I would love to know more about it, maybe some reference docs?

Or if there is any example we can directly look into, that would be even better 🙏
